### PR TITLE
fix(ui): replace tsup with tsc in dev command

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Remove `tsup` in preference of `tsc` for dev commands

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -81,7 +81,6 @@
     "lint-staged": ">=10",
     "prettier": "2.4.1",
     "rimraf": "^3.0.2",
-    "tsup": "^5.11.9",
```
Comment (reesscot) [inline package.json:84]: woohoo!!!

File: packages/ui/src/index.ts [hunk 2]
```diff
@@ -3,3 +3,4 @@ export * from './i18n';
 export * from './machines';
 export * from './theme';
 export * from './types';
+export const hehe = '';
```
[the remark this answers is not in the document]
Comment (reesscot) [reply packages/ui/src/index.ts:6]: my favorite const ever

[the remark this answers is not in the document]
Comment (wlee221) [reply packages/ui/src/index.ts:6]: This might necessitate a API review 😉

File: packages/react-core/tsconfig.dev.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    // rollup does not allow `declarationDir` to be assigned in the tsconfig
```
Comment (reesscot) [inline packages/react-core/tsconfig.dev.json:4]: What does this mean? Looks like this is in the tsconfig?

Comment (calebpollman) [reply packages/react-core/tsconfig.dev.json:4]: The dev tsconfig extends from the base tsconfig

File: packages/react/tsconfig.json [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
     "declaration": true,
     "esModuleInterop": true,
-    "jsx": "react-jsx",
-    "rootDir": "./src",
-    "outDir": "./dist",
     "importHelpers": true,
+    "jsx": "react",
+    "module": "esnext",
+    "moduleResolution": "node",
+    "outDir": "dist/esm",
```
Comment (reesscot) [inline packages/react/tsconfig.json:10]: Assuming this is ignored by rollup?

File: packages/ui/tsconfig.json [hunk 5]
```diff
@@ -5,25 +5,21 @@
     }
   },
   "compilerOptions": {
-    "target": "es5",
```
Comment (reesscot) [inline packages/ui/tsconfig.json:8]: Can you confirm that the only change here was outDir? Hard to read with the alphabetizing

File: packages/react/tsconfig.json (hunk 4 above)
Comment (calebpollman) [reply packages/react/tsconfig.json:10]: It aligns to our `rollup` config (has the same value as `dir`) -> [source](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/rollup.config.js#L29)

File: packages/ui/tsconfig.json (hunk 5 above)
Comment (calebpollman) [reply packages/ui/tsconfig.json:8]: That's correct

File: packages/react-core/tsconfig.dev.json (hunk 3 above)
Comment (reesscot) [reply packages/react-core/tsconfig.dev.json:4]: ok, then maybe change the comment to say 
```
 // rollup does not allow `declarationDir` to be assigned in the base tsconfig
 ```

**Comment by reesscot** [comment]
If you can make the two changes above, I'm ready to approve

File: packages/ui/src/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (zchenwei) [reply packages/ui/src/index.ts:6]: I like it

**Review by zchenwei — COMMENTED** [review]
LGTM. Thanks for working on it!

File: packages/ui/tsconfig.json [hunk 6]
```diff
@@ -5,25 +5,21 @@
...
-    "declaration": true,
-    "outDir": "dist",
-    "baseUrl": ".",
-    "types": ["webpack-env", "jest"],
-    "paths": {
-      "@/*": ["src/*"]
-    },
-    "lib": ["esnext", "dom", "dom.iterable", "scripthost"]
+    "strict": false,
+    "target": "es5",
```
Comment (zchenwei) [inline packages/ui/tsconfig.json:21]: Thinking if we should update the target to align other consumer packages. Those customer facing packages target a higher ES version which makes es5 looks dumb here.

Comment (calebpollman) [reply packages/ui/tsconfig.json:21]: Agree but believe that update has the potential to be a breaking change

**Review by reesscot — APPROVED** [review]
LGTM

**Pull request closed without merging**